presto-main/src/main/java/io/prestosql/execution/TaskStatus.java

```diff
@@ -223,6 +230,12 @@ public Duration getFullGcTime()
         return fullGcTime;
     }
 
+    @JsonProperty
+    public Map<String, Domain> getDynamicFilterDomains()
```

Comment:
Any reason not use a `TupleDomain` instead of breaking it up by column?

Reply:
`Map<String, Domain>` contain `Domains` for different joins. `TupleDomain` will mark entire domain as none if one `Domain` object is none. This would not be a correct behavior.

There was a bug actually about this

Reply:
Is the key of the map a dynamic filter id or a column name?

Reply:
> Is the key of the map a dynamic filter id or a column name?

filter id

Reply:
Ok, I'm confused then. From the point of view of the split manager, etc, a dynamic filter is an expression over multiple columns. Here, we're representing a filter over a single column. At what point are we breaking them apart and re-joining them? Maybe we're just mixing up concepts and need different terminology.

Reply:
1. Task might contain multiple joins `A, B`
2. Join `A` might have build side empty. Join `B` might have build side not empty
3. A `Map<String, Domain>` is sent back to coordinator with dynamic filters. It will contain empty domains for join `A`.
4. There are table scans `I` and `J`. `I` can be filtered on `A` symbols. `J` can be filtered on `B` symbols
5. `TupleDomains` are constructed from `Map<String, Domain>`  for table scans `I` and `J`. `TupleDomain` for TS `I` will be empty.